prospero-cli/src/main/java/org/wildfly/prospero/cli/commands/channel/ChannelAddCommand.java

```diff
@@ -20,61 +20,95 @@
 import org.wildfly.channel.Channel;
 import org.wildfly.channel.ChannelManifestCoordinate;
 import org.wildfly.channel.Repository;
+import org.wildfly.channel.maven.ChannelCoordinate;
 import org.wildfly.prospero.actions.MetadataAction;
 import org.wildfly.prospero.api.ArtifactUtils;
+import org.wildfly.prospero.api.MavenOptions;
+import org.wildfly.prospero.api.ProvisioningDefinition;
 import org.wildfly.prospero.api.RepositoryUtils;
-import org.wildfly.prospero.cli.ActionFactory;
+import org.wildfly.prospero.api.TemporaryFilesManager;
+
+import org.wildfly.prospero.cli.ChannelUtils;
+import org.wildfly.prospero.cli.ArgumentParsingException;
 import org.wildfly.prospero.cli.CliConsole;
+import org.wildfly.prospero.cli.ActionFactory;
 import org.wildfly.prospero.cli.CliMessages;
 import org.wildfly.prospero.cli.RepositoryDefinition;
 import org.wildfly.prospero.cli.ReturnCodes;
-import org.wildfly.prospero.api.TemporaryFilesManager;
 import org.wildfly.prospero.cli.commands.AbstractCommand;
 import org.wildfly.prospero.cli.commands.CliConstants;
 import picocli.CommandLine;
 
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
 import java.nio.file.Path;
 import java.util.List;
 import java.util.Optional;
 
 @CommandLine.Command(name = CliConstants.Commands.ADD, sortOptions = false)
 public class ChannelAddCommand extends AbstractCommand {
 
-    @CommandLine.Option(names = CliConstants.CHANNEL_NAME, required = true)
+    @CommandLine.Option(names = CliConstants.CHANNEL_NAME)
     private String channelName;
 
-    @CommandLine.Option(names = CliConstants.CHANNEL_MANIFEST, required = true)
-    private String gavUrlOrPath;
+    @CommandLine.Option(names = CliConstants.CHANNEL_MANIFEST)
+    private String manifestLocation;
 
     @CommandLine.Option(names = CliConstants.DIR)
     private Optional<Path> directory;
 
-    @CommandLine.Option(names = CliConstants.REPOSITORIES, split = ",", paramLabel = CliConstants.REPO_URL, required = true)
+    @CommandLine.Option(names = CliConstants.REPOSITORIES, split = ",", paramLabel = CliConstants.REPO_URL)
     private List<String> repositoryDefs;
 
+    @CommandLine.Option(names = CliConstants.CHANNEL)
+    private String channelLocation;
+
+
     public ChannelAddCommand(CliConsole console, ActionFactory actionFactory) {
         super(console, actionFactory);
     }
 
     @Override
     public Integer call() throws Exception {
         Path installationDirectory = determineInstallationDirectory(directory);
+        List<Channel> channels = null;
+        if (!channelLocation.isEmpty()) {
+            final ChannelCoordinate coord = ArtifactUtils.channelCoordFromString(channelLocation);
```

Comment:
I'm not sure if this is needed. If the location is simple URL or a path to the channel file, we don't need the whole ArtifactUtils.channelCoordFromString/ChannelUtils.resolveChannels thing, it should be enough to read the file using ChannelMapper.

Handling a case where the location is a GA[V] is more complex, but I'm not sure if that's something we want to be doing.